src/expandableCalendar/index.tsx

```diff
@@ -384,22 +384,21 @@ const ExpandableCalendar = (props: ExpandableCalendarProps) => {
       _height.current = toValue || newValue;
       _isOpen = _height.current >= threshold; // re-check after _height.current was set
 
+      resetWeekCalendarOpacity(_isOpen);
       Animated.spring(deltaY, {
         toValue: _height.current,
         speed: SPEED,
         bounciness: BOUNCINESS,
         useNativeDriver: false
-      }).start();
-
-      onCalendarToggled?.(_isOpen);
-
-      setPosition(() => _height.current === closedHeight ? Positions.CLOSED : Positions.OPEN);
+      }).start(() => {
+        onCalendarToggled?.(_isOpen);
+        setPosition(() => _height.current === closedHeight ? Positions.CLOSED : Positions.OPEN);
+      });
       closeHeader(_isOpen);
-      resetWeekCalendarOpacity(_isOpen);
     }
   };
 
-  const resetWeekCalendarOpacity = (isOpen: boolean) => {
+  const resetWeekCalendarOpacity = async (isOpen: boolean) => {
```

Comment:
Why async if no one awaits?

Reply:
It appears that `resetWeekCalendarOpacity` is causing a minor delay and occasional fluctuations in the animation, likely due to blocking the main thread. As a result, both processes are now being executed simultaneously.